### DISCUSSION TITLE
When should usePress be used, given that useButton exists?

### DISCUSSION THREAD

**Discussion by samselikoff** [adobe/react-spectrum#3314] [Q&A] [answered]
Hey there, amazing library + documentation!

I'm curious about what the use case is for `usePress`. It seems that anything "pressable" should be a button, and therefore `useButton` should be used for the additional features and semantics it provides.

Could someone give some advice around what the appropriate use cases are for `usePress`?

Thanks!

**Comment by reidbarber** [comment, marked as answer]
Good question! It usually wouldn't be used by itself, but more likely would be used for handling press interaction within other ARIA patterns. For instance, a radio button requires press interaction, but doesn't have button semantics, so our `useRadio` hook uses it.

If you search the codebase for `usePress`, you'll see where it gets used within other hooks.

**Reply by snowystinger** [reply]
Just to add to that, we use it to implement all of our menu items, picker items, and table rows, which also are not buttons. Those are all implemented by `useSelectableItem` https://github.com/adobe/react-spectrum/blob/main/packages/%40react-aria/selection/src/useSelectableItem.ts#L267

**Reply by samselikoff** [reply]
I see. Thank you both for such a quick response!

So just to be absolutely sure I'm thinking about it right, if I were making a button using React Aria, `useButton` would always be the way to go – correct?

**Reply by reidbarber** [reply]
Correct! Definitely go with `useButton`.

**Reply by snowystinger** [reply]
if it should be `role=button` or is a button element then you should consider using `useButton`, but there may be exceptions. For example, we use `usePress` to implement our `useCalendarCell`'s buttonProps which has a role 'button' but is more complex than your ordinary button. A good rule of thumb though is that if it's a button, you should start by using `useButton` and then if usability testing shows some deficiencies, you can log an issue and contribute a fix here or you can implement what you need using `usePress`.

**Reply by samselikoff** [reply]
Makes sense. Thanks again for all the info, really loving this project + appreciate all the work you all have put into it! 🍻
